internal/protocol/message.go

```diff
@@ -229,12 +217,46 @@ func (m *Message) putNamedValues(values NamedValues) {
 			panic("unsupported value type")
 		}
 	}
+}
+
+// Encode the given driver values as binding parameters.
+func (m *Message) putNamedValues(values NamedValues) {
+	l := len(values)
+	if l > 255 {
+		// safeguard, should have been checked beforehand.
+		panic("too many parameters")
+	}
+
+	n := uint8(l) // N of params
+	if n == 0 {
```

Comment:
Somehow this early return on `len(values) == 0` would feel more natural sooner. Would it be better to have it just before the `l > 255` check?

The same would hold for `Message.putNamedValues32`

Reply:
I don't think it can happen earlier, because the point of the `l > 255` check is to be sure that the `n := uint8(l)` cast is safe.

Reply:
I'm not sure I follow, the `> 255` check makes the `!= 0` check look independent to me; is this not possible?
```go
l := len(values)
if l == 0 {
    return
}
if l > 255 {
    panic("...")
}

m.putUint8(uint8(l))
```